react/react1/week3/homework/todo-list/src/components/TodoList.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from "react";
+import TodoListItem from "./TodoListItem.js";
+import dobby from "../data/dobby.gif";
+import AddTodo from "./AddTodo.js";
+import Border from "./Border";
+import PropTypes from "prop-types";
+
+export default function TodoList({ fetchUrl }) {
+  const [todoList, setTodoList] = useState([]);
+
+  useEffect(() => {
+    async function fetchTodo() {
+      const response = await fetch(fetchUrl);
+      const fetchedTodo = await response.json(response);
+      setTodoList(fetchedTodo);
+    }
+    fetchTodo();
+  }, [fetchUrl]);
+
+  function addTodo(description, deadline) {
+    if (description === "" && deadline === null) {
+      alert("Please, add some description and a deadline for your todo!");
+      return;
+    } else {
+      let todoId;
+      if (todoList.length === 0) {
+        todoId = 1;
+      } else {
+        todoId = todoList[todoList.length - 1].id + 1;
+      }
+      console.log(todoList);
+      return setTodoList((todoList) => [
+        ...todoList,
+        { id: todoId, description, deadline },
+      ])
+    };
+  }
+
+
+  //I have a problem with updating value (toDoList). My handleUpdate function gets a description from the input but we can only see the result of updating in console and not in the rendered list
+  const handleUpdate = (id, description) => {
+    const updatedTodoList = [...todoList].map((todo) => {
```

Comment:
Not sure what the original format of the response you are getting is that is being put into todoList. But if it is already an array then [...todoList] should be pointless since todoList will be same as [...todoList] 

If it's an object you can ignore this comment :)